File: pharmacy-backend/src/main/java/com/harshit/pharmacy/prescription/service/PrescriptionService.java
package com.harshit.pharmacy.prescription.service;

import com.harshit.pharmacy.prescription.dto.PrescriptionResponse;
import com.harshit.pharmacy.prescription.dto.UploadPrescriptionRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

public interface PrescriptionService {

    PrescriptionResponse uploadPrescription(MultipartFile prescription, UploadPrescriptionRequest request);

    Page<PrescriptionResponse> getMyPrescriptions(Pageable pageable);

    PrescriptionResponse replacePrescription(Integer prescriptionId, MultipartFile file);

    PrescriptionResponse getPrescriptionById(Integer prescriptionId);

}

File: pharmacy-frontend/src/services/prescriptionService.js
import api from "../api/api";

export async function getMyPrescriptions({ page, size }) {
    const response = await api.get("/prescriptions", {
        params: { page, size }
    });

    return response.data;
}

export async function uploadPrescription(file, doctorName) {
    const formData = new FormData();

    formData.append("prescription", file);
    formData.append("doctorName", doctorName);

    const response = await api.post("/prescriptions", formData);

    return response.data;
}

export async function replacePrescription(prescriptionId, file) {
    const formData = new FormData();

    formData.append("prescription", file);

    const response = await api.put(
        `/prescriptions/${prescriptionId}`,
        formData
    );

    return response.data;
}

export async function getPrescriptionById(prescriptionId) {
    const response = await api.get(`/prescriptions/${prescriptionId}`);

    return response.data;
}